h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -222,14 +234,24 @@ public MP getParams() {
     public ModelParametersBuilderFactory<MP> getParametersBuilderFactory() {
       return _paramsBuilderFactory;
     }
-
-    protected MP getModelParams(MP params, Object[] hyperParams) {
+    
+    private Map<String, Object[]>[] extractSubspaces() {
+      if(!_hyperParams.containsKey(SUBSPACES)) { return new Map[0]; }
+      return Stream.of(_hyperParams.get(SUBSPACES)).toArray(Map[]::new);
```

Comment:
ok, I think I know why you need this strange conversion: see my suggestion in h2o-core/src/main/java/hex/schemas/GridSearchSchema.java
The original array was created as an `Object[]` instead of a`Map[]`, so the Java interpreter fails casting it to `Map[]` later. You have to create it as a `Map[]`, then it is casted as `Object[]`: casting is mainly useful for the compiler, not the interpreter, so that you'll then be able to read it directly and cast it here:
```suggestion
      return (Map<>[])_hyperParams.get(SUBSPACES));
```

Reply:
Had to cast it to ```Map<String, Object[]>[]```, but this seemed to work. Thanks for the suggestion.